### PR TITLE
Move amtool to modular structure

### PR DESCRIPTION
**Pull request by stuartnelson3**
implements #1297

### CODE REVIEW

File: cli/alert.go [hunk 1]
```diff
@@ -14,16 +14,21 @@ import (
...
+	matcherGroups     []string
+}
 
-func init() {
-	alertQueryCmd.Action(queryAlerts)
+func configureAlertCmd(app *kingpin.Application, longHelpText map[string]string) {
+	var (
+		a             = &alertQueryCmd{}
+		alertCmd      = app.Command("alert", "View and search through current alerts")
+		alertQueryCmd = alertCmd.Command("query", "View and search through current alerts").Default()
```
Comment (simonpasquier) [inline cli/alert.go]: I find it a bit confusing to have both `alertQueryCmd` as a local variable here and as a global struct.

File: cli/config_resolver.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
+package cli
```
Comment (simonpasquier) [inline cli/config_resolver.go]: I've already noticed that most of the files in the `cli` package are missing the Apache 2 license header. This shouldn't be addressed in this PR but rather tackled once https://github.com/prometheus/prometheus/issues/3904 is fixed.

**Comment by stuartnelson3** [comment]
@simonpasquier any more comments/fixes?

File: cli/check_config.go [hunk 3]
```diff
@@ -8,23 +8,28 @@ import (
...
-)
+// TODO: This can just be a type that is []string, doesn't have to be a struct
+type checkConfigCmd struct {
+	files []string
+}
+
+func configureCheckConfigCmd(app *kingpin.Application, longHelpText map[string]string) {
+	var (
+		c              = &checkConfigCmd{}
+		checkConfigCmd = app.Command("check-config", "Validate alertmanager config files")
```
Comment (simonpasquier) [inline cli/check_config.go]: Same confusing names with the struct declared at L12.

**Comment by simonpasquier** [comment]
:+1:

**Pull request merged**